shopfloor_single_product_transfer_same_next_picking/services/single_product_transfer.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+# from functools import wraps
+
+from odoo.addons.component.core import Component
+
+
+class ShopfloorSingleProductTransfer(Component):
+    _inherit = "shopfloor.single.product.transfer"
+
+    def _set_quantity__check_location(self, location, move_line, confirmation=False):
+        res = super()._set_quantity__check_location(location, move_line, confirmation)
+        # Could also be asking for confirmation with a warning
+        if res and res.get("message", {}).get("message_type", "") == "error":
+            return res
+        if not location.quant_ids:
+            return res
+        lines = self.env["stock.move.line"].search(
+            [
+                ("location_id", "child_of", location.ids),
+                (
+                    "state",
+                    "in",
+                    (
+                        "assigned",
+                        "partially_available",
+                    ),
+                ),
+            ]
+        )
+        if not lines:
+            return res
+        # From the move line to work on get the picking for its destination moves
+        same_picking = move_line.move_id.move_dest_ids.picking_id
+        # In the destination there should be only moves from that picking.
+        if all(move in same_picking.move_lines for move in lines.move_id):
+            return res
```

Comment:
This should be in a module in the backend that will ensure the constrain on action done. The integration in shopfloor should be on top of that one to anticipate the error but where the logic is